### PR TITLE
chore: Clean-up dangerfiles and add AB# check

### PR DESCRIPTION
**Pull request by jasmineHundal**
Fixes: AB#30629

- Remove `Big-PR` checks and move changelog check to `incent-api` and `incent-config-gem`, to be removed when `release-please` is implemented.
- Add check if `AB#` is present in PR body for feature branches (in gems).
- Make` incent-config ` danger checks flexible to allow for conventional commits; `type:(incent-config-repo)` format

### CODE REVIEW

File: incent-config/Dangerfile [hunk 1]
```diff
@@ -54,10 +54,10 @@ def program_set_from_branch
 end
 
 def program_set_from_commit
-  return 'na' if commit_subject.include?('[na]') || commit_body.include?('[na]')
+  return 'na' if not_applicable_subject? || not_applicable_body?
 
   @program_set_from_commit ||= directories.find do |dir|
-    commit_subject.include?("[#{dir}]") || commit_body.include?("[#{dir}]")
+    commit_subject.include?("[#{dir}]") || commit_body.include?("[#{dir}]") || commit_subject.include?("(#{dir})") || commit_body.include?("(#{dir})")
```
Comment (jasmineHundal) [inline incent-config/Dangerfile:60]: Getting spoilt without any rubocop length checks

File: incent-config/Dangerfile [hunk 2]
```diff
@@ -89,19 +89,19 @@ def commit_body
...
 end
 
 def not_applicable_subject?
-  commit_subject.include?('[na]')
+  commit_subject.include?('[na]') || commit_subject.include?('(na)')
 end
 
 def not_applicable_body?
-  commit_body.include?('[na]')
+  commit_body.include?('[na]') || commit_body.include?('(na)')
```
Comment (johnallen3d) [inline incent-config/Dangerfile:104]: Possible refactor here (extracting this logic). Also, maybe the extracted method could have logic along these lines (if you think it's readable).

```rb
["[na]", "(na)"].any? { |token| "foo".include?(token) }
```

**Review by johnallen3d — CHANGES_REQUESTED** [review]
Looks good @jasmineHundal. A couple of suggestions.

File: incent-config/Dangerfile [hunk 3]
```diff
@@ -97,11 +100,11 @@ def not_applicable_branch_name?
 end
 
 def not_applicable_subject?
-  commit_subject.include?('[na]') || commit_subject.include?('(na)')
+  ["[na]", "(na)"].any? { |na| commit_subject.include?(na) }
 end
 
 def not_applicable_body?
-  commit_body.include?('[na]') || commit_body.include?('(na)')
+  ["[na]", "(na)"].any? { |na| commit_body.include?(na) }
```
[the remark this answers is not in the document]
Comment (jasmineHundal) [reply incent-config/Dangerfile:107]: That one had a `starts_with?` for the first one so I left it out at first, but it the same thing in this context so doesn't matter

**Review by johnallen3d — APPROVED** [review]

**Pull request merged**